### DISCUSSION TITLE
How to modify the client default server address in the code

### DISCUSSION THREAD

**Discussion by WeiYea** [rustdesk/rustdesk#2236] [General]
Self built server，
How to modify the client default server address in the code。

自己搭建服务器，但是每次发给别人都需要去改服务器地址，修改代码的那个位置，可以默认服务器ip，以后发给别人都可以不用添加服务器地址？

**Comment by djex** [comment]
Look in the `libs/hbb_common/src/config.rs` file and change `RENDEZVOUS_SERVERS` to your own server. Make sure to also change `RS_PUB_KEY` to your server public key.

If you have your server set to only accept connections that supply a public key using the `-k _ ` flag you will need to force the client to write the `RS_PUB_KEY` to the config or else I found the server will reject the connection. My guess is because the official RustDesk servers do not enforce the public key with the `-k _ `flag. This is sort of a hack but what I did was this:

In `libs/hbb_common/src/config.rs` ->inside ` impl Config2 `-> inside `fn load() `-> add the line `config.options.insert("key".to_string(), RS_PUB_KEY.to_string());`

**Reply by WeiYea** [reply]
fucking great！
I tried it. It was very effective. Thank you very much.
Do you know how to package an executable program into a single file that does not need to be installed, similar to the official "rustdesk-1.1.9-windows_x64-portable"

Sorry, I use Chinese and translate it into English, which may be ambiguous

**Reply by djex** [reply]
Currently the build.py script will only builds portable version for flutter UI. I am not sure sure how they built the 1.1.9 portable version using sciter UI. The released source for 1.1.9 is different than the 1.1.9 download version on the website.

What I did was I stuck with the sciter UI for now since flutter UI is not complete yet.  I used the existing code to embed the sciter.dll file into the compiled exe and then extract sciter.dll on program run to the same location as the exe. This keeps it portable to a point however configuration files will still be created in the user AppData folders. It's not a truly portable solutions but it works. If this is what your looking for I can explain a bit more.

**Reply by WeiYea** [reply]
Thank you. I have found a way on the Internet.

Another problem: I modified the default server address in the new version, but the server configuration interface still displays the address manually set before. Is this because there is a hidden configuration file? Or does the system registry record the relevant configuration? How can I not read the previously configured server address?

**Reply by djex** [reply]
You should clear the old config files. There are two locations RustDesk will save the configs.

C:\Users\USER_NAME\AppData\Roaming\RustDesk
C:\Windows\ServiceProfiles\LocalService\AppData\Roaming\RustDesk

Delete the config folders and restart RustDesk
